ex1/exceptions.py

```diff
@@ -46,3 +72,22 @@
 # the first by the second. This function should handle exceptions that might
 # occur print out what went wrong and return None if no results could be
 # computed.
+def safe_divide(a1,a2):
+	"""
+	Parameters
+	----------
+	a1,a2 : int
+
+	Returns
+	-------
+	double
+		if dividing is possible
+
+	None
+		by dividing error
+    	"""
+	try:
+		rez = a1/a2
+		return rez
+	except:
```

Comment:
Excepting all possible exceptions is generally a dangerous idea. Because you can not really distinguish anymore why something went wrong.

In more complex cases than this one there is also the possibility that a Error occurs that you did not think of. If you then except all Exceptions you will never see this unforeseen error that might have side effects you did not foresee.